src/app/clusters/zone-management-server/zone-management-server.h

```diff
@@ -0,0 +1,356 @@
+                             Optional<CharSpan> aColor)
+    {
+        Set(aName, aUse, aVertices, aColor);
+    }
+
+    TwoDCartesianZoneStorage(const TwoDCartesianZoneStorage & aTwoDCartZone) { *this = aTwoDCartZone; }
+
+    TwoDCartesianZoneStorage & operator=(const TwoDCartesianZoneStorage & aTwoDCartZone)
+    {
+        Set(aTwoDCartZone.name, aTwoDCartZone.use, aTwoDCartZone.verticesVector, aTwoDCartZone.color);
+        return *this;
+    }
+
+    void Set(const CharSpan & aName, ZoneUseEnum aUse, const std::vector<TwoDCartesianVertexStruct> & aVertices,
+             Optional<CharSpan> aColor)
+    {
+        nameString     = std::string(aName.begin(), aName.end());
+        name           = CharSpan(nameString.c_str(), nameString.size());
+        use            = aUse;
+        verticesVector = aVertices;
+        vertices       = DataModel::List<TwoDCartesianVertexStruct>(verticesVector.data(), verticesVector.size());
+        if (aColor.HasValue())
+        {
+            colorString = std::string(aColor.Value().begin(), aColor.Value().end());
+            color       = MakeOptional(CharSpan(colorString.c_str(), colorString.size()));
+        }
+        else
+        {
+            colorString.clear();
+            color = NullOptional;
+        }
+    }
+
+    std::string nameString;
+    std::string colorString;
+    std::vector<TwoDCartesianVertexStruct> verticesVector;
+};
+
+struct ZoneInformationStorage : ZoneInformationStruct
+{
+    ZoneInformationStorage(){};
+
+    ZoneInformationStorage(const uint16_t & aZoneID, ZoneTypeEnum aZoneType, ZoneSourceEnum aZoneSource,
+                           const Optional<TwoDCartesianZoneStorage> & aTwoDCartZoneStorage)
+    {
+        Set(aZoneID, aZoneType, aZoneSource, aTwoDCartZoneStorage);
+    }
+
+    ZoneInformationStorage(const ZoneInformationStorage & aZoneInfoStorage) { *this = aZoneInfoStorage; }
+
+    ZoneInformationStorage & operator=(const ZoneInformationStorage & aZoneInfoStorage)
+    {
+        Set(aZoneInfoStorage.zoneID, aZoneInfoStorage.zoneType, aZoneInfoStorage.zoneSource, aZoneInfoStorage.twoDCartZoneStorage);
+        return *this;
+    }
+
+    void Set(const uint16_t & aZoneID, ZoneTypeEnum aZoneType, ZoneSourceEnum aZoneSource,
+             const Optional<TwoDCartesianZoneStorage> & aTwoDCartZoneStorage)
+    {
+        zoneID              = aZoneID;
+        zoneType            = aZoneType;
+        zoneSource          = aZoneSource;
+        twoDCartZoneStorage = aTwoDCartZoneStorage;
+        twoDCartesianZone   = twoDCartZoneStorage;
+    }
+
+    Optional<TwoDCartesianZoneStorage> twoDCartZoneStorage;
+};
+
+/** @brief
+ *  Defines interfaces for implementing application-specific logic for various aspects of the ZoneManagement Cluster.
+ *  Specifically, it defines interfaces for the command handling and loading of the allocated streams.
+ */
+class ZoneMgmtDelegate
```

Comment:
So why do we have ZoneManagement::ZoneMgmtDelegate?  Why not just ZoneManagement::Delegate?